op-node/node/node.go

```diff
@@ -288,6 +293,21 @@ func (n *OpNode) initRuntimeConfig(ctx context.Context, cfg *Config) error {
 	return nil
 }
 
+func (n *OpNode) initL1BeaconAPI(ctx context.Context, cfg *Config) error {
+	if cfg.Beacon == nil {
```

Comment:
This could create a silent error and possible panic downstream on misconfiguration?

Reply:
it would certainly pose a problem if L1 beacon wasn't specified after the Ecotone upgrade. Not sure what the best way to handle this is, as I don't want to break current deploys that aren't even aware of the pending upgrade and won't be specifying the beacon client?

Reply:
Had a similar situation in Teku with the merge where it started requiring an execution engine endpoint - we wound up making the endpoint required if the merge upgrade was scheduled for the current network (even if its not yet active).  That way you can upgrade without changes if you're using a network that doesn't have Ecotone scheduled yet, but when you update to a version that is "Ecotone-ready" it fails to start unless you provide the endpoint to prevent you from thinking you're ready when you're not.

If you only require it when Ecotone is actually active, people will think they're ready and then have it fail when the upgrade actually upgrades.  They're paying more attention when they do an upgrade so that's the time to make it fail. :)

Reply:
added an error log message here to make them aware of the pending upgrade if it's not specified.

Reply:
ok changed based on @ajsutton feedback to error out if the upgrade is scheduled but no endpoint is specified

Reply:
Good feedback @ajsutton!

Reply:
```
if cfg.Beacon == nil {
		n.log.Error("No beacon endpoint configured. Configuration is mandatory for the Ecotone upgrade.")
		return nil
	}
```

Reply:
thank you, not sure what happened there. fixed.

Reply:
[edit]  This is done in the "check" function in config.go.